lib/src/utils.dart

```diff
@@ -15,28 +15,63 @@
 import 'dart:io';
 
 import 'package:logging/logging.dart';
+import 'package:yaml/yaml.dart';
+
+export 'package:yaml/yaml.dart' show loadYaml;
 
 final RegExp importExportPackageRegex =
     new RegExp(r'''^(import|export)\s+['"]package:([a-zA-Z_]+)\/.+$''', multiLine: true);
 
-const dependenciesKey = 'dependencies';
-const dependencyValidatorPackageName = 'dependency_validator';
-const devDependenciesKey = 'dev_dependencies';
-const nameKey = 'name';
-const transformersKey = 'transformers';
-
 final Logger logger = new Logger('dependency_validator');
 
 String bulletItems(Iterable<String> items) => items.map((l) => '  * $l').join('\n');
 
 Iterable<File> listDartFilesIn(String dirPath) {
   if (!FileSystemEntity.isDirectorySync(dirPath)) return const [];
-  return new Directory(dirPath)
-      .listSync(recursive: true)
-      .where((entity) => entity is File && !entity.path.contains('/packages/') && entity.path.endsWith('.dart'));
+
+  final list = new Directory(dirPath).listSync(recursive: true)
+    ..retainWhere((entity) => entity is File && !entity.path.contains('/packages/') && entity.path.endsWith('.dart'));
+
+  return new List<File>.from(list);
 }
 
 void logDependencyInfractions(String infraction, Iterable<String> dependencies) {
   final sortedDependencies = dependencies.toList()..sort();
   logger.warning([infraction, bulletItems(sortedDependencies), ''].join('\n'));
 }
+
+class PubspecYaml {
+  final dynamic _yamlMap;
+  static const _dependenciesKey = 'dependencies';
+  static const _dependencyValidatorPackageName = 'dependency_validator';
+  static const _devDependenciesKey = 'dev_dependencies';
+  static const _nameKey = 'name';
+  static const _transformersKey = 'transformers';
+  static const _pubspecPath = 'pubspec.yaml';
+
+  PubspecYaml() : _yamlMap = loadYaml(new File(_pubspecPath).readAsStringSync());
+
+  String get name => _yamlMap[_nameKey] as String;
+
+  Set<String> get dependencies =>
+      ((_yamlMap[_dependenciesKey] as YamlMap ?? const <dynamic, dynamic>{}).keys as Iterable<String>).toSet();
+
+  Set<String> get devDependencies =>
+      ((_yamlMap[_devDependenciesKey] as YamlMap ?? const <dynamic, dynamic>{}).keys as Iterable<String>).toSet()
+        // Remove this package, since we know they're using our executable
+        ..remove(_dependencyValidatorPackageName);
```

Comment:
Filtering out the `dependency_validator` package from the dev deps seems like it should happen elsewhere, since this is just a generic `PubspecYaml` class to help parse a `pubspec.yaml` file.

Reply:
Hmm, I meant this class to be specific to this use-case. That's why I made it private. I can rename it if you'd like

Reply:
yeah I guess it's private so it's not a big deal